Doc/library/asyncio-task.rst

```diff
@@ -1067,14 +1067,59 @@ Scheduling From Other Threads
    This function is meant to be called from a different OS thread
    than the one where the event loop is running.  Example::
 
-     # Create a coroutine
-     coro = asyncio.sleep(1, result=3)
-
-     # Submit the coroutine to a given loop
-     future = asyncio.run_coroutine_threadsafe(coro, loop)
-
-     # Wait for the result with an optional timeout argument
-     assert future.result(timeout) == 3
+     def in_thread(loop: asyncio.AbstractEventLoop) -> None:
+         # Run some blocking IO
+         pathlib.Path("example").read_text(encoding="utf8")
+
+         # Create a coroutine
+         coro = asyncio.sleep(1, result=3)
+
+         # Submit the coroutine to a given loop
+         future = asyncio.run_coroutine_threadsafe(coro, loop)
+
+         # Wait for the result with an optional timeout argument
+         assert future.result(timeout=2) == 3
+
+     async def amain() -> None:
+         # Get the running loop
+         loop = asyncio.get_running_loop()
+
+         # Run something in a thread
+         await asyncio.to_thread(in_thread, loop)
+
+   It's also possible to run the other way around.  Example::
+
+     @contextlib.contextmanager
+     def loop_in_thread() -> Generator[asyncio.AbstractEventLoop]:
```

Comment:
FWIW I think the above simpler example is sufficient, this one looks very contrived to me involving thread pool executor and all.

Reply:
It's used to get any failures out, if they happen when constructing the loop

Reply:
I meant that can we avoid mixing concurrent futures with loop? Starting loop in a thread should be sufficient here no?  I don't want to encourage such code which creates loop in another thread but runs code from another thread.

Reply:
You need to be able to start and stop the thread, and collect errors that get returned from asyncio.run, it's complicated to do that with threading.Thread

Reply:
I am thinking of something like this:
```py
from threading import Thread
import contextlib
import asyncio
import concurrent.futures


@contextlib.contextmanager
def get_loop():
    loop_fut = concurrent.futures.Future()
    stop = asyncio.Event()

    async def runner():
        loop_fut.set_result(asyncio.get_event_loop())
        await stop.wait()

    try:
        t = Thread(target=lambda: asyncio.run(runner()))
        t.start()
        loop = loop_fut.result()
        yield loop
    finally:
        loop.call_soon_threadsafe(stop.set)
        t.join()


with get_loop() as loop:
    r = asyncio.run_coroutine_threadsafe(
        asyncio.sleep(1, result="hello"), loop
    ).result()
    print(r)

```

Reply:
But ThreadPoolExecutor(1) does all that for you, and collects any result from asyncio.run. Eg your code will hang if the OS has run out of file descriptors, mine will propagate the exception

Reply:
Okay I take it, I am not really a fan of this pattern but I am fine if you wish.